### ISSUE TITLE
Contracts documentation does not explain where to put the ABI library

### ISSUE THREAD

**Issue by paulrberg** [FuelLabs/sway#3432]
To quote the docs on [Syntax of a Smart Contract](https://fuellabs.github.io/sway/v0.31.1/sway-program-types/smart_contracts.html#syntax-of-a-smart-contract):

> It is considered good practice to define your ABI in a separate library and import it into your contract.

This is great. I love finding out what the best practices are right from the get-go.

However, there is no explanation for the file structure. Should it be something like this?

```
├── lib.sw # ABI library
└── main.sw # contract
```

I tried the above but it didn't seem to work.

There is also no explanation for how the library should be imported in the contract file.

**Comment by mohammadfawaz** [comment]
Ahh good point. The idea here is to move the ABI into a separate _library_ project. We will make sure this is properly documented, thanks!

**Comment by paulrberg** [comment]
Thanks @mohammadfawaz. It might also be worth it switch the order of the contracts with the libraries, like this:

- 3 Program Type
    - 3.1 Libraries
    - 3.2 Contracts

This is because the contracts docs rely upon the user knowing how to handle libraries.

**Comment by mohammadfawaz** [comment]
Thanks @paulrberg - Good suggestion. The tasks here can all be done together.

**Comment by paulrberg** [comment]
A follow-up question on this topic, @mohammadfawaz.

Is there something equivalent to [Cargo workspaces](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html) in Forc/ Sway?

**Comment by otrho** [comment]
Yes! https://fuellabs.github.io/sway/v0.32.2/book/forc/workspaces.html

**Comment by paulrberg** [comment]
Awesome! Thanks, @otrho.
